test/unit/math/test_ad_matvar.hpp

```diff
@@ -251,129 +251,119 @@ inline void test_matvar_gradient(const ad_tolerances& tols, ResultMV& A_mv_ret,
   }
 }
 
+/** @name convert_to_matvar
+ * Convert input to be matvar compatible with the
+ * same scalar type as T.
+ */
+///@{
 /**
- * Check that if at least one of the input types is a var with inner
- * Eigen type that the output is not an Eigen matrix of vars.
+ * The scalar type of T is artithmetic, so just pass the input through.
  *
- * Similarly check that if none of the inputs are vars with inner type
- * Eigen that the output is not a var with inner Eigen type.
- *
- * @tparam ReturnType Type of result of calculation
- * @tparam Types Types of input arguments
- * @param ret not used, only types are used.
- * @param x not used, only types are used.
- */
-template <typename ReturnType, typename... Types,
-          require_all_not_std_vector_t<ReturnType, Types...>* = nullptr,
-          require_any_matrix_t<Types...>* = nullptr>
-void check_return_type(const ReturnType& ret, const Types&... x) {
-  using stan::is_eigen;
-  using stan::is_var_matrix;
-  using stan::math::var_value;
-  using stan::math::test::type_name;
-  if (stan::math::conjunction<is_eigen<Types>...>::value
-      && !(is_eigen<ReturnType>::value
-           || std::is_same<ReturnType, var_value<double>>::value)) {
-    FAIL() << "All the arguments are `Eigen::Matrix<T, R, C>` types, but the "
-              "return type is neither an `Eigen::Matrix<T, R2, C2>` or a `var`";
-  } else if (stan::math::disjunction<is_var_matrix<Types>...>::value
-             && !(is_var_matrix<ReturnType>::value
-                  || std::is_same<ReturnType, var_value<double>>::value)) {
-    FAIL() << "One of the arguments is a `var_value<Eigen::Matrix<double, R, "
-              "C>>`, but the return type is not a `var_value<T>`";
-  }
+ * @tparam T Target type
+ * @tparam S Argument type
+ * @param x argument
+ * @return x
+ */
+template<typename T, typename S,
+	 require_all_st_arithmetic<T, S>* = nullptr>
+auto convert_to_matvar(const S& x) {
+  return x;
 }
 
 /**
- * Check that if the input type is a `std::vector` of vars with inner
- * Eigen type that the output is not a `std::vector` of Eigen matrices of vars.
+ * The scalar type of T is var here, so make a copy of
+ * x with scalar type promoted to var.
  *
- * Similarly check that if the input is a `std::vector` of Eigen matrices of
- * vars that the output is not a `std::vector` of vars with inner Eigen type.
- *
- * @tparam ReturnType Element types of result of calculation
- * @tparam Types Element types of calculation input
- * @param ret not used, only types are used.
- * @param x not used, only types are used.
- */
-template <typename ReturnType, typename Type, require_matrix_t<Type>* = nullptr>
-void check_return_type(const std::vector<ReturnType>& ret,
```

Comment:
Why did this get deleted?

Reply:
Replaced by https://github.com/stan-dev/math/blob/ab230e663626f038ccdc30030cac52475e4ffdad/test/unit/math/test_ad_matvar.hpp#L426 (as discussed on Slack)